Source/Diagnostics/ParticleDiag/ParticleDiag.cpp

```diff
@@ -0,0 +1,71 @@
+#include "WarpX.H"
+#include "ParticleDiag.H"
+// #include "Utils/WarpXUtil.H"
+
+#include <AMReX_ParmParse.H>
+
+using namespace amrex;
+
+ParticleDiag::ParticleDiag(std::string diag_name, std::string name, WarpXParticleContainer* pc)
+    : m_diag_name(diag_name), m_name(name), m_pc(pc)
+{
+    ParmParse pp(diag_name + "." + name);
+    if (!pp.queryarr("variables", variables)){
+        variables = {"Ex", "Ey", "Ez", "Bx", "By", "Bz", "ux", "uy", "uz", "w"};
+    }
+
+    //variable to set plot_flags size
+    int plot_flag_size = PIdx::nattribs;
+    if(WarpX::do_back_transformed_diagnostics && m_pc->doBackTransformedDiagnostics())
+        plot_flag_size += 6;
+
+#ifdef WARPX_QED
+    if(m_pc->DoQED()){
+        // plot_flag will have an entry for the optical depth
+        plot_flag_size++;
+    }
+#endif
+
+    // Set plot_flag to 0 for all attribs
+    plot_flags.resize(plot_flag_size, 0);
+
+    // If not none, set plot_flags values to 1 for elements in plot_vars.
+    if (variables[0] != "none"){
+        for (const auto& var : variables){
+            // Return error if var not in PIdx.
+            WarpXUtilMsg::AlwaysAssert(
+                ParticleStringNames::to_index.count(var),
+                "ERROR: variables argument '" + var +
+                "' not in ParticleStringNames"
+                );
+            plot_flags[ParticleStringNames::to_index.at(var)] = 1;
+        }
+    }
+
+#ifdef WARPX_DIM_RZ
+    // Always write out theta, whether or not it's requested,
+    // to be consistent with always writing out r and z.
+    plot_flags[ParticleStringNames::to_index.at("theta")] = 1;
+#endif
+
+#ifdef WARPX_QED
+    if(m_pc->DoQED()){
+        //Optical depths is always plotted if QED is on
+        plot_flags[plot_flag_size-1] = 1;
```

Comment:
this looks dangerous, if someone adds another attribute.
would a `map` work better for `plot_flags`?

Reply:
Thanks for pointing it out, I agree it is misleading, dangerous and also probably wrong (if both QSR and BW are on, I expect this to cause a bug). A solution would be to do the same as for other variables above in the same file, e.g.,
```c++
ParticleStringNames::to_index.at("opt_depth_BW")
ParticleStringNames::to_index.at("opt_depth_breit_QSR")
```
but I do not think these exist. So I think the problem is general and should be fixed in the particle containers first. I can open an issue for that.

Reply:
sounds good, can you just add a TODO or warning note here, so one can grep for this?